ec2deprecateimg

```diff
@@ -31,311 +31,383 @@ from ec2imgutils.ec2imgutilsExceptions import (
+    )
+    parser.add_argument(
+        '--verbose',
+        action='store_true',
+        default=False,
+        dest='verbose',
+        help='Enable on verbose output'
+    )
+    parser.add_argument(
+        '--version',
+        action='version',
+        version=utils.get_version(),
+        help='Program version'
+    )
+    parsed_args = parser.parse_args(args)
+    return parsed_args
 
-for region in regions:
-    deprecator.set_region(region)
+
+# ----------------------------------------------------------------------------
+def deprecate_images_in_region(deprecator, dryRun, region, errors, logger):
+    """Function that deprecates the specified images for a AWS region. The
+    function returns the errors dictionary with any error it found added
+    """
     try:
-        if args.dryRun:
+        if dryRun:
             logger.info('Dry run, image attributes will not be modified')
             deprecator.print_deprecation_info()
         else:
             deprecator.deprecate_images()
     except EC2DeprecateImgException as e:
         errors[region] = format(e)
     except Exception as e:
-        if errors:
-            logger.error('Collected errors:')
-            for region, error in list(errors.items()):
-                logger.error('Region: %s -> %s' % (region, error))
+        print_errors(errors, logger)
         logger.exception(e)
         sys.exit(1)
+    return errors
+
+
+# ----------------------------------------------------------------------------
+def print_errors(errors, logger):
+    """Function that prints the collected errors during image deprecation
+    process
+    """
+    if errors:
+        logger.error('Collected errors:')
+        for region, error in list(errors.items()):
+            logger.error('Region: %s -> %s' % (region, error))
+
+
+# ----------------------------------------------------------------------------
+def main(args):
+    args = parse_args(args)
+    logger = utils.get_logger(args.verbose)
+    check_required_arguments(args, logger)
+
+    config = get_config(args, logger)
+    access_key = get_access_key(args, config, logger)
+    secret_key = get_secret_key(args, config, logger)
+
+    regions = utils.get_regions(args, access_key, secret_key)
+    deprecator = get_deprecator(args, access_key, secret_key, logger)
+
+    # Collect all the errors to be displayed later
+    errors = {}
+    for region in regions:
+        deprecator.set_region(region)
+        errors = deprecate_images_in_region(
+            deprecator,
+            args.dryRun,
+            region,
+            errors,
+            logger
+        )
+
+    if errors:
+        print_errors(errors, logger)
+        sys.exit(1)
 
-if errors:
-    logger.error('Errors encountered:')
-    for region, error in list(errors.items()):
-        logger.error('Region: %s -> %s' % (region, error))
-    sys.exit(1)
+
+# ----------------------------------------------------------------------------
+if __name__ == '__main__':
```

Comment:
See the other suggestion. This makes switching to entry_points console_scripts possible.

Reply:
No runtime dependency on the lousy setuptools implementation. I know the Python community has adopted the setuptools dependency as the defacto standard. Never the less it is lousy to have to depend on a whole library implementation that is principally for development to generate an executable.

Using the `if __name__ == '__main__':` is fine.

Reply:
Not sure if I'm following you completely on the setuptools topic. You prefer to leave the scripts in the 'scripts' section of the setup.py rather that including them as entry_points-console_scripts (to avoid the setuptools wrapping), right?

Let me prepare the other PR with the scripts refactor including functions (and some tests) and let's discuss  further steps from there, ok? 
(the goal of the refactor was just to ease the testing of the rpm build for the package)